header/verify.go

```diff
@@ -23,43 +38,41 @@ func (eh *ExtendedHeader) Verify(untrusted libhead.Header) error {
 	}
 
 	isAdjacent := eh.Height()+1 == untrst.Height()
-	if isAdjacent {
-		// Optimized verification for adjacent headers
-		// Check the validator hashes are the same
-		if !bytes.Equal(untrst.ValidatorsHash, eh.NextValidatorsHash) {
-			return &libhead.VerifyError{
-				Reason: fmt.Errorf("expected old header next validators (%X) to match those from new header (%X)",
-					eh.NextValidatorsHash,
-					untrst.ValidatorsHash,
-				),
-			}
-		}
-
-		if !bytes.Equal(untrst.LastHeader(), eh.Hash()) {
-			return &libhead.VerifyError{
-				Reason: fmt.Errorf("expected new header to point to last header hash (%X), but got %X)",
-					eh.Hash(),
-					untrst.LastHeader(),
-				),
-			}
-		}
-
+	if !isAdjacent {
+		// if the header is non-adjacent, it will be verified later via
+		// VerifyAdjacent once its adjacent (n-1) header is synced
```

Comment:
We should make a TODO here and, ideally link to the issue. That's a temporary fix., as we want to have skipping verification long term